src/overlays/actors/ovl_En_Wonder_Item/z_en_wonder_item.c

```diff
@@ -26,25 +47,316 @@ const ActorInit En_Wonder_Item_InitVars = {
     (ActorFunc)EnWonderItem_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Item/EnWonderItem_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Wonder_Item/func_80B380A0.s")
+Vec3f sTagPointsFree[9];
+Vec3f sTagPointsOrdered[9];
+
+void EnWonderItem_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnWonderItem* this = THIS;
+
+    if ((this->collider.dim.radius != 0) || (this->collider.dim.height != 0)) {
+        Collider_DestroyCylinder(globalCtx, &this->collider);
+    }
+}
+
+void func_80B380A0(EnWonderItem* this, GlobalContext* globalCtx, s32 autoCollect) {
+    static s16 dropTable[] = {
+        ITEM00_NUTS,        ITEM00_HEART_PIECE,  ITEM00_MAGIC_LARGE,   ITEM00_MAGIC_SMALL,
+        ITEM00_HEART,       ITEM00_ARROWS_SMALL, ITEM00_ARROWS_MEDIUM, ITEM00_ARROWS_LARGE,
+        ITEM00_RUPEE_GREEN, ITEM00_RUPEE_BLUE,   ITEM00_RUPEE_RED,     ITEM00_FLEXIBLE,
+    };
+    s32 i;
+    s32 randomDrop;
+
+    func_80078884(NA_SE_SY_GET_ITEM);
+
+    if (this->dropCount == 0) {
+        this->dropCount++;
+    }
+    for (i = this->dropCount; i > 0; i--) {
+        if (this->itemDrop < WONDERITEM_RANDOM) {
+            if ((this->itemDrop == WONDERITEM_FLEXIBLE) || !autoCollect) {
+                Item_DropCollectible(globalCtx, &this->actor.posRot.pos, dropTable[this->itemDrop]);
+            } else {
+                Item_DropCollectible(globalCtx, &this->actor.posRot.pos, dropTable[this->itemDrop] | 0x8000);
+            }
+        } else {
+            randomDrop = this->itemDrop - WONDERITEM_RANDOM;
+            if (!autoCollect) {
+                Item_DropCollectibleRandom(globalCtx, NULL, &this->actor.posRot.pos, randomDrop);
+            } else {
+                Item_DropCollectibleRandom(globalCtx, NULL, &this->actor.posRot.pos, randomDrop | 0x8000);
+            }
+        }
+    }
+    if (this->switchFlag >= 0) {
+        Flags_SetSwitch(globalCtx, this->switchFlag);
+    }
+    Actor_Kill(&this->actor);
+}
+
+void EnWonderItem_Init(Actor* thisx, GlobalContext* globalCtx) {
+    static u32 collisionTypes[] = {
+        WONDERITEM_DAMAGE_SLASH,     WONDERITEM_DAMAGE_ARROW,     WONDERITEM_DAMAGE_HAMMER,   WONDERITEM_DAMAGE_BOMB,
+        WONDERITEM_DAMAGE_SLINGSHOT, WONDERITEM_DAMAGE_BOOMERANG, WONDERITEM_DAMAGE_HOOKSHOT,
+    };
+    s32 pad;
+    s16 colTypeIndex;
+    EnWonderItem* this = THIS;
+    s16 rotZover10;
+    s16 tagIndex;
+
+    osSyncPrintf("\n\n");
+    // Mysterious mystery
+    osSyncPrintf("\x1b[32m☆☆☆☆☆ 不思議不思議まか不思議 \t   ☆☆☆☆☆ %x\n\x1b[m", this->actor.params);
```

Comment:
Don't forget to use the VT macros! There's a few other places in the file too. :)

Reply:
Handled.